test/jdk/java/nio/channels/DatagramChannel/Promiscuous.java

```diff
@@ -222,8 +208,8 @@ public static void main(String[] args) throws IOException {
             InetAddress source = config.ip4Addresses(nif).iterator().next();
             test(INET, nif, ip4Group1, ip4Group2);
 
-            // Solaris and Linux allow IPv6 sockets join IPv4 multicast groups
-            if (os.equals("Linux"))
+            // Linux allows IPv6 sockets join IPv4 multicast groups
+            if (Platform.isLinux())
```

Comment:
If you can, it would be useful to double check it is Linux specific, I would expect it should pass on Windows and macOS too.